samples/notebooks/prediction/SDK_Pytorch_Custom_Predict.ipynb

```diff
@@ -0,0 +1,1603 @@
+    "        Returns:\n",
+    "            The preprocessed prediction input.\n",
+    "        \"\"\"\n",
+    "        return prediction_input\n",
+    "\n",
+    "    def predict(self, instances: Any) -> Any:\n",
+    "        \"\"\"Performs prediction.\n",
+    "        Args:\n",
+    "            instances (Any):\n",
+    "                Required. The instances to perform prediction.\n",
+    "        Returns:\n",
+    "            Prediction results.\n",
+    "        \"\"\"\n",
+    "        raise NotImplementedError(\"Predictor.predict has not been implemented yet.\")\n",
+    "\n",
+    "    def postprocess(self, prediction_results: Any) -> Any:\n",
+    "        \"\"\"Postprocesses the prediction results.\n",
+    "        Args:\n",
+    "            prediction_results (Any):\n",
+    "                Required. The prediction results.\n",
+    "        Returns:\n",
+    "            The postprocessed prediction results.\n",
+    "        \"\"\"\n",
+    "        return prediction_results\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "-KODNAqK5k8R"
+   },
+   "source": [
+    "First, implement a custom `Predictor` that loads in both the preprocesor and the model. The preprocessor and the model will then be used at `predict` time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "fkh8DT7CCgQO"
+   },
+   "outputs": [],
+   "source": [
+    "%%writefile $USER_SRC_DIR/predictor.py\n",
+    "\n",
+    "import pandas as pd\n",
+    "import pickle\n",
+    "import torch\n",
+    "\n",
+    "from google.cloud import storage\n",
+    "from google.cloud.aiplatform.prediction.predictor import Predictor\n",
+    "\n",
+    "\n",
+    "class CprPredictor(Predictor):\n",
+    "    \n",
+    "    def __init__(self):\n",
+    "        self._class_names = [\"setosa\", \"versicolor\", \"virginica\"]\n",
+    "    \n",
+    "    def load(self, gcs_artifacts_uri: str):\n",
+    "        \"\"\"Loads the preprocessor and model artifacts.\"\"\"\n",
+    "        model_file = \"model.pt\"\n",
+    "        \n",
+    "        gcs_client = storage.Client()\n",
+    "        with open(model_file, 'wb') as model_f:\n",
+    "            gcs_client.download_blob_to_file(\n",
+    "                f\"{gcs_artifacts_uri}/{model_file}\", model_f\n",
+    "            )\n",
+    "\n",
+    "        self._model = torch.load(model_file)\n",
+    "\n",
+    "    def predict(self, instances):\n",
+    "        \"\"\"Performs prediction.\"\"\"\n",
+    "        instances = instances[\"instances\"]\n",
+    "        data = pd.DataFrame(instances).values\n",
+    "        inputs = torch.Tensor(data)\n",
+    "        outputs = self._model(inputs)\n",
+    "        _ , predicted = torch.max(outputs, 1)\n",
+    "        return {\"predictions\": [self._class_names[class_num] for class_num in predicted]}"
```

Comment:
These should go in the postprocess method. Make sure it's clear what this is doing (find the index with maximum value from the class probabilities produced by the module, then return the class name)

Reply:
Done.